packages/client/components/EditorLinkChanger/EditorLinkChangerDraftjs.tsx

```diff
@@ -37,11 +51,49 @@ const EditorLinkChangerDraftjs = (props: Props) => {
   } = props
   useTaskChild('editor-link-changer')
   const handleSubmit = ({text, href}: {text: string; href: string}) => {
+    const keepSelection = true
+    // don't include leading and trailing whitespace in the link text
+    const nonWhitespaceFromStart = text.search(/\S/)
+    const nonWhitespaceFromEnd = text.search(/\s*$/)
+    const startStr = nonWhitespaceFromStart === -1 ? '' : text.slice(0, nonWhitespaceFromStart)
+    const endStr = nonWhitespaceFromEnd === -1 ? '' : text.slice(nonWhitespaceFromEnd)
+    const trimmedText = text.trim() ? text.trim() : text
```

Comment:
-1 could you please add a comment why you're doing this
```ts
    // trim link text if it contains any non-whitespace characters, otherwise keep it verbatim
```
could also be shortened to
```ts
    const trimmedText = text.trim() || text
```
if you like